cmd/interval/add/add.go

```diff
@@ -15,60 +15,181 @@
 package add
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"github.com/edgexfoundry-holding/edgex-cli/pkg/editor"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/scheduler"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/urlclient/local"
 	"io/ioutil"
 
 	"github.com/edgexfoundry-holding/edgex-cli/config"
-	request "github.com/edgexfoundry-holding/edgex-cli/pkg"
 
 	"github.com/edgexfoundry/go-mod-core-contracts/clients"
 	"github.com/edgexfoundry/go-mod-core-contracts/models"
 
-	"github.com/pelletier/go-toml"
 	"github.com/spf13/cobra"
 )
 
-type IntervalFile struct {
-	Intervals []models.Interval
+var interactiveMode bool
+var name string
+var start string
+var end string
+var cron string
+var runOnce bool
+var frequency string
+
+var file string
+
+const IntervalTempl = `[{{range $interval := .}}` + baseIntervalTemp +
+	`{{end}}]`
+
+const baseIntervalTemp = `{
+	"Name" : "{{.Name}}",
+	"start" : "{{.Start}}",
+	"end" : "{{.End}}",
+	"cron" : "{{.Cron}}",
+	"runOnce" : {{.RunOnce}},
+	"frequency" : "{{.Frequency}}"
 }
+`
 
 func NewCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "add",
 		Short: "Add interval",
-		Long:  `Create the intervals described in the given TOML files.`,
-		Run:   addIntervalHandler,
+		Long:  `Create interval(s) described in the given JSON file or use the interactive mode with additional flags.`,
+		RunE:  newIntervalHandler,
 	}
+	cmd.Flags().BoolVarP(&interactiveMode, editor.InteractiveModeLabel, "i", false, "Open a default " +
```

Comment:
s/a default/the default

Reply:
I will include this fix within my next PR